src/uu/split/src/split.rs

```diff
@@ -783,6 +807,29 @@ impl Settings {
                 }
             }
         }
+
+        // Make sure that separator is only one UTF8 character (if specified)
+        // defaults to '\n' - newline character
+        // If the same separator (the same value) was used multiple times - `split` should NOT fail
+        // If the separator was used multiple times but with different values (not all values are the same) - `split` should fail
+        let separator = match matches.get_many::<String>(OPT_SEPARATOR) {
+            Some(sep_values) => match sep_values.map(|s| s.to_string()).reduce(|cur, nxt| {
+                if cur == nxt {
+                    nxt
+                } else {
+                    cur + &nxt
+                }
+            }) {
+                Some(s) => match s.as_str() {
+                    "\\0" => b'\0',
+                    _ if s.as_bytes().len() == 1 => s.as_bytes()[0],
+                    _ => return Err(SettingsError::MultiCharacterSeparator(s.to_owned())),
```

Comment:
There's a subtlety here in the GNU implementation:
```
> split -t1 -t2
split: multiple separator characters specified
> split -t12
split: multi-character separator ‘12’
```
I don't really mind this version though. I just wanted to note it's not quite the same.

Reply:
in latest refactor - added new error variant + tests to distinguish the two